### PR TITLE
Create hint design

### PR DESCRIPTION
**Pull request by thomastepi**
- Replaced the static preview with previewComponent prop.
- created hint preview component.

### CODE REVIEW

**Review by erenfn — CHANGES_REQUESTED** [review]
This breaks the current implementation of the Popup Preview Component. Make the necessary changes so that both the Popup and Hint components work, and ensure it accommodates future components like Tours (although the design is not yet finished).

**Comment by gorkem-bwl** [comment]
Hi @thomastepi - any updates here?

**Comment by thomastepi** [comment]
> Hi @thomastepi - any updates here? 
in progress. will push today @gorkemcetin

File: frontend/src/scenes/login/Login.module.css [hunk 1]
```diff
@@ -47,7 +47,7 @@
 .form-group input[type="password"],
 .form-group input[type="surname"] {
   width: 100%;
-  padding: 8px 12px;
+  /* padding: 8px 12px; */
```
Comment (erenfn) [inline frontend/src/scenes/login/Login.module.css:50]: why did you comment this out?

**Review by erenfn — CHANGES_REQUESTED** [review]
Selecting popup size as large causes overflows.

image

Also we can get rid of the grey part at the bottom in the popup

Ekran görüntüsü 2024-11-06 144832

**Review by uparkalau — CHANGES_REQUESTED** [review]
Review Coderabbit recommendation. Resolve that suggestions what make sense.

**Comment by gorkem-bwl** [comment]
> Fixed overflow and removed the grey area at the bottom. For large popup sizes, should the dialog maintain a fixed width with horizontal scroll or expand/shrink based on content width? @erenfn

Let's expand/shrink the popup based on the content width.

**Review by erenfn — CHANGES_REQUESTED** [review]
- In hint preview, these buttons on the bottom are over each other
image
- In create new popup content, I think the 'Editor' and 'preview' buttons are not used to calculate the modal size. That's why they don't fit the screen adn we get the grey area at the bottom
image
- Also after creating/editing a popup/hint the default page doesn't fetch the new popup/hint automatically to show it on the screen
image

**Comment by erenfn** [comment]
I previously set the position of the 'Editor' and 'Preview' buttons to absolute. This was because, when you change the size of the popup or hint, I wanted those buttons to stay in place and not get pushed. You can change it if you want. Or @gorkem-bwl should be move them to the top of the right section instead of the bottom?

File: frontend/src/products/Hint/HintComponent.jsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            }}
+          >
+            <h3>{header}</h3>
+          </div>
+        )}
+        <div
+          className="preview-content"
+          style={{ color: textColor }}
+          dangerouslySetInnerHTML={{ __html: DOMPurify.sanitize(content) }}
+        />
```
[the remark this answers is not in the document]
Comment (erenfn) [reply frontend/src/products/Hint/HintComponent.jsx:34]: What do you think about this suggestion?

**Comment by gorkem-bwl** [comment]
> I previously set the position of the 'Editor' and 'Preview' buttons to absolute. This was because, when you change the size of the popup or hint, I wanted those buttons to stay in place and not get pushed. You can change it if you want. Or @gorkem-bwl should be move them to the top of the right section instead of the bottom?

I think it's good as is. When the size of the popup changes they should stay in place.

**Comment by erenfn** [comment]
Then maybe we can add a `maxHeight` (of the preview component) prop to `RichTextEditor` to calculate where we should place the editor and preview buttons

**Review by erenfn — CHANGES_REQUESTED** [review]
Looks good! After you change the `CreateBanner` route like popup, and resolve the conflicts I think it is ready to merge. 

Also @gorkem-bwl Should we set the minimum height of the popup to the maximum height of the content and appearance sections? Switching between the two could cause the popup height to change, which might be eye-straining.

image

Ekran görüntüsü 2024-11-16 191001

**Comment by gorkem-bwl** [comment]
> Also @gorkem-bwl Should we set the minimum height of the popup to the maximum height of the content and appearance sections? Switching between the two could cause the popup height to change, which might be eye-straining.

Yep, that makes sense. Overall the design looks good otherwise.

**Review by gorkem-bwl — CHANGES_REQUESTED** [review]
Minor changes requested.

**Comment by erenfn** [comment]
Created #401 with the merged changes

**Pull request closed without merging**